pkg/realizer/runnable/realizer.go

```diff
@@ -147,40 +149,39 @@ func (p *runnableRealizer) Realize(ctx context.Context, runnable *v1alpha1.Runna
 	return stampedObject, outputs, nil
 }
 
-func resolveSelector(ctx context.Context, selector *v1alpha1.ResourceSelector, repository repository.Repository, namespace string) (map[string]interface{}, error) {
+func (r *runnableRealizer) resolveSelector(ctx context.Context, selector *v1alpha1.ResourceSelector, repository repository.Repository, discoveryClient discovery.DiscoveryInterface, namespace string) (map[string]interface{}, error) {
 	log := logr.FromContextOrDiscard(ctx)
 
 	if selector == nil {
 		return nil, nil
 	}
 
-	results, err := repository.ListUnstructured(ctx, schema.FromAPIVersionAndKind(selector.Resource.APIVersion, selector.Resource.Kind), namespace, selector.MatchingLabels)
+	apiResourceList, err := discoveryClient.ServerResourcesForGroupVersion(selector.Resource.APIVersion)
 	if err != nil {
-		log.Error(err, "failed to list objects matching selector", "selector", selector.MatchingLabels)
-		return nil, fmt.Errorf("failed to list objects matching selector [%+v]: %w", selector.MatchingLabels, err)
+		log.Error(err, "failed to list server api resources")
+		return nil, fmt.Errorf("failed to list server api resources: %w", err)
 	}
 
-	if len(results) == 0 {
-		log.V(logger.DEBUG).Info("selector did not match any objects, checking resolveClusterScopedSelector")
-		return resolveClusterScopedSelector(ctx, selector, repository)
-	} else if len(results) > 1 {
-		log.V(logger.DEBUG).Info("selector matched multiple objects")
-		return nil, fmt.Errorf("selector matched multiple objects")
+	var namespaced bool
+	for _, apiResource := range apiResourceList.APIResources {
+		if apiResource.Kind == selector.Resource.Kind {
+			namespaced = apiResource.Namespaced
```

Comment:
should we break out of the loop given that we've found that it's namespaced?

Reply:
good catch- missed that when I was refactoring